numpy/core/src/multiarray/compiled_base.c

```diff
@@ -415,6 +417,22 @@ arr_insert(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwdict)
         goto fail;
     }
     numvals = PyArray_SIZE(avals);
+
+    if (!numvals) {
+        /* nothing to insert! fail unless none of mask is true */
+        PyObject *const any = PyArray_Any(amask, NPY_MAXDIMS, NULL);
+        const int fail = any == PyArrayScalar_True;
```

Comment:
Hmm, does `any` need to be checked for `NULL`? At least should use `Py_XDECREF`. Be best to put `(...)` around the logical expression on the right for clarity.